x/oracle/keeper/aggregator/calculator.go

```diff
@@ -66,6 +66,31 @@ type roundPricesList struct {
 	roundPricesCount int
 }
 
+func (r *roundPricesList) copy4CheckTx() *roundPricesList {
+	ret := &roundPricesList{
+		roundPricesList:  make([]*roundPrices, len(r.roundPricesList)),
+		roundPricesCount: r.roundPricesCount,
+	}
+
+	for _, v := range r.roundPricesList {
+		tmpRP := &roundPrices{
+			detID:     v.detID,
+			price:     big.NewInt(v.price.Int64()),
+			prices:    make([]*priceAndPower, len(v.prices)),
+			timestamp: v.timestamp,
+		}
+		for _, pNP := range v.prices {
+			tmpPNP := *pNP
+			// power will be modified during execution
+			tmpPNP.power = big.NewInt(pNP.power.Int64())
```

Comment:
Is overflow for this checked elsewhere? I know it's illogical for it to happen but it should be guarded whenever price is originally loaded.

Reply:
fixed